keel-clouddriver/src/main/kotlin/com/netflix/spinnaker/keel/clouddriver/model/ActiveServerGroup.kt

```diff
@@ -7,6 +7,57 @@ import com.netflix.spinnaker.keel.clouddriver.CloudDriverCache
 import com.netflix.spinnaker.keel.core.api.Capacity
 import com.netflix.spinnaker.kork.exceptions.SystemException
 
+data class ServerGroupCollection(
+  val accountName: String,
+  val serverGroups: Set<ServerGroup>
+)
+
+/**
+ * Objects that are returned when querying for all fo the server groups associated with a cluster.
+ *
+ * Two differences from [ActiveServerGroup]:
+ *   - disabled flag
+ *   - no accountName field (since this is defined on the parent [ServerGroupCollection] object
```

Comment:
I feel like there's an opportunity here to take advantage of that `disabled` flag and merge those two types into one.

Reply:
@luispollo What I'm struggling with for unifying them is that I want to have a ServerGroup class that has a non-nullable `accountName` field.